sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Helpers/AnalysisContextExtensions.cs

```diff
@@ -53,7 +53,7 @@ public static void ReportDiagnosticWhenActive(this CodeBlockAnalysisContext cont
 
         private static void ReportWhenNotSuppressed(Diagnostic diagnostic, Action<Diagnostic> report)
         {
-            if (!SonarAnalysisContext.ShouldDiagnosticBeReported(diagnostic))
+            if (SonarAnalysisContext.ShouldDiagnosticBeReported(diagnostic))

```

Comment:
Oh gosh! We shall have caught this during review.

Reply:
I changed the negation in the name, hence the tests are reversed :-/